lib/SQL/Translator/Producer/MySQL.pm

```diff
@@ -457,11 +457,17 @@ sub create_table
     #
     my @constraint_defs;
     my @constraints = $table->get_constraints;
+
+    # Get the primary key, if there is one. No need to create an index for an
+    # FK that is also the PK. It may be a multi-field key, so generate a
+    # suitable string for comparison.
+    my $pk = join "\0", map $_->fields, grep $_->type eq PRIMARY_KEY, @constraints;
```

Comment:
Here you're joining all the fields in the primary key, but below you're testing that against the first field in the foreign key, so this can't work at all for composite primary keys.

I don't if the current handling is actually correct for compound indexes and foreign keys, but either way, I think primary keys should be handled the same as normal indexes (i.e. just add each field to `%indexed_fields`).